app/javascript/shared/components/useKeyboardShortcuts.jsx

```diff
@@ -44,6 +44,44 @@ function isFormField(element) {
  *
  */
 export function useKeyboardShortcuts(shortcuts, eventTarget = window) {
+  const [keyChain, setKeyChain] = useState([]);
+  const [keyChainQueue, setKeyChainQueue] = useState(null);
+
+  // Work out the correct shortcut for the key press
+  const callShortcut = useCallback((e, keys) => {
+    let shortcut;
+    if (keyChain.length > 0) {
+      shortcut = shortcuts[`${keyChain.join("~")}~${e.code}`];
+    } else {
+      shortcut = shortcuts[`${keys}${e.code}`] || shortcuts[`${keys}${e.key.toLowerCase()}`];
+    }
+
+    // if a valid shortcut is found call it and reset the chain
+    if (shortcut) {
+      shortcut(e);
+      setKeyChain([]);
+    }
+  }, [shortcuts, keyChain]);
+
+  // Set up key chains
+  useEffect(() => {
+    if (!keyChainQueue && keyChain.length === 0) return;
+    let timeout;
+
+    timeout = window.setTimeout(() => {
+      clearTimeout(timeout);
+      setKeyChain([]);
+    }, 500);
```

Comment:
would it make sense to have this configurable? If I'm not mistaken in reading the code, GitHub uses 1.5 seconds, I did a non scientific experiment and indeed hotkeys are rvalid for a while :D

https://github.com/github/hotkey/blob/0691586c0eca5d198b5619b7321594706f5cc194/src/index.ts#L22

Reply:
As in make is variable set in the backend? Or are you thinking we pull it from the env file?